src/spring-cloud/azext_spring_cloud/app.py

```diff
@@ -171,6 +172,12 @@ def app_update(cmd, client, resource_group, service, name,
         'https_only': https_only,
     }
 
+    if deployment is None:
+        updated_deployment_kwargs = {k: v for k, v in deployment_kwargs.items() if v}
+        if updated_deployment_kwargs:
+            raise InvalidArgumentValueError('{} cannot be set when there is no active deployment.'
```

Comment:
Is `ArgumentUsageError` a more appropriate error type for this usage case?

Reply:
updated